tsdb/engine/tsm1/data_file.go

```diff
@@ -855,6 +856,60 @@ func (t *TSMReader) Key(index int) string {
 	return t.index.Key(index)
 }
 
+func (t *TSMReader) Next(key string, timestamp time.Time) ([]Value, error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	entries := t.index.Entries(key)
+	var entry *IndexEntry
+
+	if timestamp.Before(entries[0].MinTime) {
+		entry = entries[0]
+	} else {
+		for i, e := range entries {
```

Comment:
This is look for the next block based on where timestamp resides.  L866 checks to see if the timestamps is before any block in the file.  If it is, we use the first block in the file.  Other wise, we find the block where the timestamp resides and use the next one after it.